airbyte-integrations/connectors/source-linnworks/source_linnworks/streams.py

```diff
@@ -70,50 +69,37 @@ def next_page_token(self, response: requests.Response) -> Optional[Mapping[str,
             }
 
 
-class Location(LinnworksStream):
-    # https://apps.linnworks.net/Api/Method/Locations-GetLocation
-    # Response: StockLocation https://apps.linnworks.net/Api/Class/linnworks-spa-commondata-Locations-ClassBase-StockLocation
+class StockLocations(LinnworksStream):
+    # https://apps.linnworks.net/Api/Method/Inventory-GetStockLocations
+    # Response: List<StockLocation> https://apps.linnworks.net/Api/Class/linnworks-spa-commondata-Inventory-ClassBase-StockLocation
     # Allows 150 calls per minute
     primary_key = "StockLocationIntId"
+    use_cache = True
 
     def path(
         self, stream_state: Mapping[str, Any] = None, stream_slice: Mapping[str, Any] = None, next_page_token: Mapping[str, Any] = None
     ) -> str:
-        return "/api/Locations/GetLocation"
-
-    def request_params(
-        self, stream_state: Mapping[str, Any], stream_slice: Mapping[str, any] = None, next_page_token: Mapping[str, Any] = None
-    ) -> MutableMapping[str, Any]:
-        return {"pkStockLocationId ": stream_state["pkStockLocationId"]}
+        return "/api/Inventory/GetStockLocations"
 
 
-class StockLocations(LinnworksStream):
-    # https://apps.linnworks.net/Api/Method/Inventory-GetStockLocations
-    # Response: List<StockLocation> https://apps.linnworks.net/Api/Class/linnworks-spa-commondata-Inventory-ClassBase-StockLocation
+class StockLocationDetails(HttpSubStream, StockLocations):
+    # https://apps.linnworks.net/Api/Method/Locations-GetLocation
+    # Response: StockLocation https://apps.linnworks.net/Api/Class/linnworks-spa-commondata-Locations-ClassBase-StockLocation
     # Allows 150 calls per minute
     primary_key = "StockLocationIntId"
 
+    def __init__(self, **kwargs):
+        super().__init__(StockLocations(**kwargs), **kwargs)
```

Comment:
You could unit test the `__init__` to make sure future eventual changes to the parent declaration makes the test fail.